### ISSUE TITLE
What can picongpu do?

### ISSUE THREAD

**Issue by baobaba13** [ComputationalRadiationPhysics/picongpu#4178]
Hi，I would like to ask, can picongpu simulate the electromagnetic pulse radiation generated by laser irradiating solid targets? That is, the laser irradiates the solid target to produce hot electrons, some of which with high energy pass through the potential barrier and become runaway electrons. The runaway electrons move in the simulated cavity to excite electromagnetic radiation pulses.

**Comment by sbastrakov** [comment]
Hello @baobaba13 , thanks for your question.

@PrometheusPi do you have an opinion on that?

**Comment by PrometheusPi** [comment]
Hi @baobaba13,
while I do not quite understand the details of your proposed setup, I would say that in general simulating the radiation from a laser irradiated foil is possible. We simulated for examples high harmonic generation (HHG)  via the relativistically oscillating mirror (ROM) effect from laser irradiated foils in the target normal sheath acceleration (TNSA) configuration. 
As long as the radiation you are interested is classical in nature, the radiation plugin is capable of simulating the emitted radiation.

**Comment by baobaba13** [comment]
@PrometheusPi Hi  PrometheusPi，First of all, thank you for your reply. The problem I described is: numerical simulation of electromagnetic pulse in the target chamber of nanosecond laser inertial confinement device. The laser target interaction process will produce a large amount of electromagnetic Pulse (EMP), which has the characteristics of short time, large amplitude change and high intensity in the time domain, covers a wide range of frequency bands in the frequency domain. It is unknown whether picongpu can conduct numerical simulation for this scenario.

**Comment by steindev** [comment]
Hey @baobaba13, PIConGPU simulates the classic interaction of charged particles and electromagnetic fields. Additionally, extensions such ionization routines can be activated to increase the fidelity in the simulation of plasmas.

In your described setup, the interaction of the laser with the particles, and the interaction of the produced radiation with the particles will be described correctly as long as your cell size is small enough to resolve the smallest expected wavelength.

For wavelengths smaller than this resolution limit, you can still calculate their spectrum with the help of the radiation plugin, but there will be no interaction between these wavelengths and the particles.

In summary, I think you should be able to obtain a decent understanding of the spectrum of the EMP using PIConGPU.

**Comment by BrianMarre** [comment]
@baobaba13 Please remember to close the issue, if your question was answered.
